### PR TITLE
PLNSRVCE-1022: add kube-rbac-proxy related rback for sprayproxy; add sprayproxy servicemonitor; bump sprayproxy git/image sha

### PR DESCRIPTION
**Pull request by gabemontero**
@adambkaplan @bamachrn @gbenhaim - changes stemming from the call between Gal, Bama, and I last week, plus the merge of https://github.com/redhat-appstudio/sprayproxy/pull/19

PR is based off of the latest multi-cluster branch

### CODE REVIEW

File: components/sprayproxy/kustomization.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  - https://github.com/redhat-appstudio/sprayproxy/config?ref=076942559b739729c156b3d64f87efda1731ab90
+  - https://github.com/redhat-appstudio/sprayproxy/config?ref=0eece5fdaf5c84ed79c852b7c9c7118ced5d4390
 
 namespace: sprayproxy
 
 images:
   - name: ko://github.com/redhat-appstudio/sprayproxy
     newName: quay.io/redhat-appstudio/sprayproxy
-    newTag: 076942559b739729c156b3d64f87efda1731ab90
+    newTag: 0eece5fdaf5c84ed79c852b7c9c7118ced5d4390
```
Comment (gabemontero) [inline components/sprayproxy/kustomization.yaml:9]: fyi this is available now per https://quay.io/repository/redhat-appstudio/sprayproxy?tab=tags

**Comment by gabemontero** [comment]
@adambkaplan - should I be added to https://github.com/redhat-appstudio/infra-deployments/blob/multi-cluster/components/sprayproxy/OWNERS ?

**Review by adambkaplan — APPROVED** [review]
/approve

For the bits around the sprayproxy component. @bamachrn does the RBAC and service monitor look right to you, given that this component will only be deployed on the host cluster?

**Comment by gabemontero** [comment]
@adambkaplan - should I be added to https://github.com/redhat-appstudio/infra-deployments/blob/multi-cluster/components/sprayproxy/OWNERS ?

File: components/sprayproxy/kustomization.yaml (hunk 1 above)
Comment (gabemontero) [reply components/sprayproxy/kustomization.yaml:9]: there is for the main branch @avinal but it does not work as I understand it for the multi cluster branch

Comment (gabemontero) [reply components/sprayproxy/kustomization.yaml:9]: and per the open PRs at https://github.com/redhat-appstudio/infra-deployments/pulls you'll see there is no PRs pending to update the sprayproxy's sha ref

Comment (avinal) [reply components/sprayproxy/kustomization.yaml:9]: Got it. I was not aware of the multi-cluster branch.

**Review by gbenhaim — APPROVED** [review]

**Pull request merged**